meson.build

```diff
@@ -158,6 +158,15 @@ if not get_option('source-only')
 
     lite_deps = [lua_dep, sdl_dep, freetype_dep, pcre2_dep, libm, libdl]
 endif
+
+#===============================================================================
+# Link to X11 on linux to detect scale factor when video driver is x11
+#===============================================================================
+if host_machine.system() == 'linux'
```

Comment:
I don't think only Linux uses X11. You might need to figure out a way to better detect this, or load it on runtime.

Reply:
Linux isn't the only system that *can* use X11 (gWSL, XQuartz, etc.) but it is mainly used on Linux.

Reply:
I guess the additional ones that would make sense are: freebsd, netbsd and openbsd as listed here https://mesonbuild.com/Reference-tables.html#operating-system-names

Reply:
Another solution:
check if X11 is available in meson.
If yes, compile the X11 code, don't link against X11 and instead load it at runtime.

This would allow systems that use X11 to use it but not make it a mandatory run-time dependency.

Reply:
ok, force pushed some changes to dynamically load the needed functionality using `SDL_LoadObject` and `SDL_LoadFunction`